broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/MigrationTransitionStep.java

```diff
@@ -48,7 +48,19 @@ public ActorFuture<Void> transitionTo(
             context.getBrokerCfg().getExperimental().getEngine().createEngineConfiguration());
 
     final var dbMigrator = new DbMigratorImpl(processingState);
-    dbMigrator.runMigrations();
+    final var transaction = zeebeDbContext.getCurrentTransaction();
+    try {
+      transaction.run(dbMigrator::runMigrations);
+      transaction.commit();
+    } catch (final Exception e) {
+      try {
+        transaction.rollback();
+      } catch (final Exception ex) {
+        return CompletableActorFuture.completedExceptionally(ex);
+      }
+      return CompletableActorFuture.completedExceptionally(e);
+    }
+
```

Comment:
💭 one though that I think should be equal and would allow us to make reuse of the run, commit or rollback logic already defined in `TransactionContext.runInTransaction`
```suggestion
    try {
      zeebeDbContext.runInTransaction(dbMigrator::runMigrations);
    } catch (final Exception e) {
      return CompletableActorFuture.completedExceptionally(e);
    }
```

if you have doubts let's follow-up on that after the release.

Reply:
I like it!